### PR TITLE
Fix streaming service resolution and header passing

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1504
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1458

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/ServerCall.java [hunk 1]
```diff
@@ -149,6 +147,14 @@ public void sendMessage(Message message) {
         }
     }
 
+    public void injectHeaders(HttpHeaders headers) {
+        if (this.outboundMessage != null) {
+            if (headers != null) {
+                headers.forEach(entry -> this.outboundMessage.setHeader(entry.getKey(), entry.getValue()));
```
Comment (chamil321) [inline grpc-native/src/main/java/org/ballerinalang/net/grpc/ServerCall.java:153]: HttpCarbonMessage has an API 
```
setHeaders(HttpHeaders httpHeaders)
```
Which can be used directly if you have exposed it in the outboundMessage. Can consider this as an improvement later

**Comment by chamil321** [comment]
@BuddhiWathsala There are some commented codes as well. please check

**Comment by BuddhiWathsala** [comment]
> @BuddhiWathsala There are some commented codes as well. please check

These commented ones are intentional. gRPC has generated codes that are repeatedly use across multiple tests. We maintain these codes as comment for the sake of completeness.

File: grpc-native/src/main/java/org/ballerinalang/net/grpc/ServerCall.java (hunk 1 above)
Comment (BuddhiWathsala) [reply grpc-native/src/main/java/org/ballerinalang/net/grpc/ServerCall.java]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/291/commits/1057107aba905b7b2e053520d9e0433e6defd5ed

**Review by daneshk — APPROVED** [review]
LGTM

**Comment by daneshk** [comment]
Let's add another test case to return ContextMessage on the server-side without using a caller object later

**Comment by BuddhiWathsala** [comment]
We already have a similar test for server streaming. I'll add similar tests for client and bidirectional.
https://github.com/ballerina-platform/module-ballerina-grpc/blob/master/grpc-ballerina/tests/32_return_record_with_headers_server_streaming_server.bal

**Pull request merged**